### PR TITLE
Disable by default layout animation

### PR DESCRIPTION
**Pull request by piaskowyk**
## Description

I added the possibility to disable layout animations. By default layout animations are disabled, but you don't need to turn it on manually, because these will be enabled if you will use the `entering`/`exiting`/`layout` prop in an animated component. You also have the possibility to disable/enable it manually.

## Example
```js
import { enableLayoutAnimations } from 'react-native-reanimated';

enableLayoutAnimations(false);
```

### CODE REVIEW

File: src/reanimated2/core.ts [hunk 1]
```diff
@@ -385,3 +385,36 @@ if (!NativeReanimatedModule.useOnlyV1) {
...
+      const userFlag = userFeaturesConfig[key];
+      if (userFlag !== false) {
+        filteredOptions[key] = option[key];
+      }
+    }
+  }
+  if (!filteredOptions) {
+    return;
+  }
+  NativeReanimatedModule.setEnableFeatures(filteredOptions);
```
Comment (piaskowyk) [inline src/reanimated2/core.ts:419]: You need to add implementation for web

Comment (piaskowyk) [reply src/reanimated2/core.ts:419]: done 👍

Comment (piaskowyk) [reply src/reanimated2/core.ts:419]: thanks

File: src/createAnimatedComponent.tsx [hunk 2]
```diff
@@ -496,6 +496,7 @@ export default function createAnimatedComponent(
           (this.props.layout || this.props.entering || this.props.exiting) &&
           tag != null
         ) {
+          setEnableFeatures({ layoutAnimation: true }, false);
```
Comment (piaskowyk) [inline src/createAnimatedComponent.tsx:499]: It can break Chrome Debugger support. Check it, please!

**Review by kmagiera — APPROVED** [review]
This looks ok, however, I think it'd be easier to add just one flag specifically for turning on layout animation as opposed to designing more generic solution to it.

If instead of `setEnableFeatures` we'd just have `enableLayoutAnimations` function we could:
1) not really care about web implementation too much. Layout Animations does not work on web so we could just no-op there
2) we now call `setEnableFeatures` on every component render. If it was just setting a single boolean variable we'd avoid additional complexity of merging and filtering dicts

**Pull request merged**